backend/webserver/views.py

```diff
@@ -70,6 +72,109 @@ def post(self, request):
+            if author.id == request.user.id:
+                post.delete()
+                return Response({"message":"Object deleted!"}, status=status.HTTP_200_OK)
+            else:
+                return Response({'message': 'You cannot delete another authors post'}, status=status.HTTP_400_BAD_REQUEST) 
+        return Response({'message': 'You can only delete public posts'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class AllPosts(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+
+    def get(self, request,pk, *args, **kwargs):
+        author = self.get_author(pk)
+        posts = author.post_set.all().order_by("-created_at")
+        serializer = PostSerializer(posts, many=True, context={'request': request})
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    
+    def post(self, request, pk, *args, **kwargs):
+        author = self.get_author(pk)
+        serializer = CreatePostSerializer(data=request.data)
+        if author.id == request.user.id:
+            if serializer.is_valid():
+                new_post = Post.objects.create(
+                    author=author,
+                    title=serializer.data["title"],
+                    description=serializer.data["description"],
+                    unlisted=serializer.data["unlisted"],
+                    content_type=serializer.data["content_type"],
+                    content=serializer.data["content"],
+                    visibility=serializer.data["visibility"]
+                )
+                if new_post.visibility == "FRIENDS" or new_post.visibility == "PUBLIC":
+                    for follow in author.followed_by_authors.iterator():
+                        with transaction.atomic():
+                            Inbox.objects.create(target_author=follow.follower,post=new_post)
+                elif new_post.visibility == "PRIVATE":
+                    private_post_serializer = SendPrivatePostSerializer(data=request.data)
+                    if private_post_serializer.is_valid():
+                        
+                        try:
+                            receiver =  Author.objects.get(pk=private_post_serializer.data['receiver']['id'])    
+                        except Author.DoesNotExist:
+                            return Response({'message': f'receiver author with id {private_post_serializer.data["receiver"]["id"]} does not exist'}, status=status.HTTP_404_NOT_FOUND)
+                        with transaction.atomic():
+                            Inbox.objects.create(target_author=receiver, post=new_post)
+                        return Response({'message': 'OK'}, status=status.HTTP_201_CREATED) 
+                    else:
+                        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
```suggestion
                    else:
                        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
You don't need line 168. It is redundant and makes things more confusing.